chore: typed jest conf + remove `version` from compose file + fix typo

kasir-barati opened the pull request:
Just some chores like getting rid of this warning in terminal each time you run e2e tests: 

```shell
WARN[0000] /home/kasir/projects/nestjs-query/examples/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
```

Typed jest conf file for better dev exp when you need to modify it.

Some typos that where here an there.

BTW I also like to ask @TriPSs why he has increased the `printWidth` of prettier to 130! I mean the recommended length is 70. So if your OK with it I am gonna change it to 70. TBH it is really long and hard to make sense of a lot of stuff. But I am also open to not touch it since it is growing on me too (as of lately I feel more comfortable around it :sweat_smile:).

One thing that bothers me though is that if I save `docker-compose.yml` or any `JSON` conf file my IDE format it and I was thinking if possible either override them in `.prettierrc` or decrease the `printWidth` to 70 since the only configuration that causes VSCode to format them in my case.

TriPSs commented on documentation/docs/introduction/example.mdx:
```diff
@@ -567,17 +569,16 @@ export class AppModule {}
 services:
   postgres:
-    image: "postgres:11.5"
+    image: "postgres:17.0-alpine3.20"
```
Think this can be just 17? No need for the alpha

kasir-barati replied:
Will test if it works with 17 (normally it would but just double checking it would not hurt :slightly_smiling_face:).

kasir-barati replied:
Done

TriPSs commented on documentation/docs/introduction/example.mdx:
```diff
@@ -567,17 +569,16 @@ export class AppModule {}
 </TabItem>
 </Tabs>
 
-Create a `docker-compose.yml` file in the root of the project
```
I can't seem to find clear documentation about compose files, can we keep docker-comoose

TriPSs replied:
Sorry, now see the link you provided.

kasir-barati replied:
No worries, just lemme know if you're happy with the new file name suggestion. If the answer is yes just resolve this conversation :slightly_smiling_face:. If not I am not really biased on naming but keeping up with what docker doc says sounds pretty good to me.

kasir-barati commented:
BTW while I was learning new stuff I saw some missing imports in the docs so I took it upon myself to fix them.

To stay relevant and up to date I removed `version` fom `docker-compose.yml` and renamed it to [`compose.yml` since it is shorter and the preferred name](https://docs.docker.com/compose/intro/compose-application-model/#the-compose-file).

Before I forget, I still had to prevent my IDE from formatting docs, otherwise there would be a really big change in this PR. So my suggestion is to add prettier as a precommit hook.

TriPSs commented:
The reason why I changed prettier width is that personally I don't like prettier at all, and the small initial width was very anying (I also do not have it enabled I think for this project on my machine 😅)

Ps: did this quick check on my phone so made some mistakes

kasir-barati commented:
> The reason why I changed prettier width is that I don't like prettier at all, and the small initial width was very anying (I also do not have it enabled I think for this project on my machine 😅)
> 
> Ps: did this quick check on my phone so made some mistakes

So what's it gonna be? We can go down 2 different path IMO:
1. Config a GH action to prettify.
2. Add `npm run format` in husky (as of our pre-commit, might slow down committing new stuff since it runs a script before each commit).

Personally I love the first solution + configured IDEs so that we can commit things fast, but we have GH workflows as some sort of guardrail. So we're gonna have a prettified codebase all the time.

TriPSs commented:
Let me think about it a bit, atm we atleast have the linter in the CI and I think my preference would go to removing prettier and do everything with eslint. But I will comeback on this.
